interop/xds/server/server.go

```diff
@@ -78,8 +87,48 @@ func (s *testServiceImpl) EmptyCall(ctx context.Context, _ *testpb.Empty) (*test
 }
 
 func (s *testServiceImpl) UnaryCall(ctx context.Context, in *testpb.SimpleRequest) (*testpb.SimpleResponse, error) {
+	response := &testpb.SimpleResponse{ServerId: s.serverID}
+
+	for _, val := range getRPCBehaviorMetadata(ctx) {
+		if st := getStatusForRPCBehaviorMetadata(val); st != nil {
+			return response, status.Err(st.Code(), st.Message())
+		}
+	}
+
+	response.Hostname = s.hostname
 	grpc.SetHeader(ctx, metadata.Pairs("hostname", s.hostname))
-	return &testpb.SimpleResponse{ServerId: s.serverID, Hostname: s.hostname}, nil
+	return response, status.Err(codes.OK, "")
+}
+
+func getRPCBehaviorMetadata(ctx context.Context) []string {
+	md, ok := metadata.FromIncomingContext(ctx)
+	if !ok {
+		logger.Error("failed to receive metadata")
+		return nil
+	}
+
+	mdRPCBehavior := md.Get(rpcBehaviorMetadataKey)
+
+	rpcBehaviorMetadata := make([]string, 0)
+	for _, val := range mdRPCBehavior {
+		splitVals := strings.Split(val, ",")
+		rpcBehaviorMetadata = append(rpcBehaviorMetadata, splitVals...)
+	}
+
+	return rpcBehaviorMetadata
+}
+
+func getStatusForRPCBehaviorMetadata(headerValue string) *status.Status {
+	if !strings.HasPrefix(headerValue, errorCodePrefix) {
+		return nil
+	}
+
+	errCode, err := strconv.Atoi(headerValue[len(errorCodePrefix):])
+	if err != nil {
+		return status.Newf(codes.InvalidArgument, "Invalid format for rpc-behavior header: %v", headerValue)
```

Comment:
This and the next one, being error strings, should not be capitalized. See: https://google.github.io/styleguide/go/decisions#error-strings